### ISSUE TITLE
Rails dupe with water bucket

### ISSUE THREAD

**Issue by ImMorpheus** [SpongePowered/Sponge#2212]
**I am currently running**

Tested with SV (https://github.com/SpongePowered/SpongeVanilla/commit/d26e6a06309f601510f871236e176a48ab0a8be8) and SF (https://github.com/SpongePowered/SpongeForge/commit/1c1d1328c13a02960ccca626fa4713f6ce43c8e0)
(SC updated to https://github.com/SpongePowered/SpongeCommon/commit/4063674e2d3f0c2c71cbf37ac7db505ba4eeeb17)

- Plugins/Mods: none

**Issue Description**

```java
@Listener
public void onChangeBlockPlace(final ChangeBlockEvent.Place event, @First Player player) {
    event.setCancelled(true);
}
``` 

1. Place water above the rail. 
2. The event is cancelled so the water is not placed and the rail is not broken, however a rail is still dropped. (duplicating the rail)

**Comment by gabizou** [comment]
Can you re-test with the following jar?

https://www.mediafire.com/file/9o40kii7x9drz1z/spongeforge-1.12.2-2768-7.1.6-RC0.jar

**Comment by gabizou** [comment]
Confirmed fixed in branch being merged in with #2219.
